analyzers/rspec/cs/S3900_c#.html

```diff
@@ -31,38 +27,55 @@ <h2>Compliant Solution</h2>
     private MyOtherClass other;
 
     public void Foo(MyOtherClass other)
-    {
-        this.other = other;
-    }
-
-    public void Bar(MyOtherClass other)
     {
         if (other != null)
         {
             this.other = other.Clone();
         }
     }
 
-    protected void FooBar(MyOtherClass other)
+    protected void Bar(MyOtherClass other)
     {
         if (other != null)
         {
             this.other = other.Clone();
         }
     }
+
+    public void Baz(MyOtherClass other)
+    {
+        ArgumentNullException.ThrowIfNull(other);
+
+        this.other = other.Clone();
+    }
+
+    public void Qux(MyOtherClass other)
+    {
+        this.other = other; // Compliant: "other" is not being dereferenced
+    }
+
+    private void Xyzzy(MyOtherClass other)
+    {
+        this.other = other; // Compliant: method is not publicly accessible
+    }
 }
 </pre>
 <h2>Exceptions</h2>
-<p>To create a custom null validation method declare an attribute with name <code>ValidatedNotNullAttribute</code> and mark the parameter that is
-validated for null in your method declaration with it:</p>
+<p>Arguments validated for <code>null</code> via helper methods should be annotated with the <a
```

Comment:
This paragraph should be probably formatted as 3 bullet points. Explaining those 3 alternative.

Currently, it reads as Microsoft, crosslinked to Reshaper followed by some custom code. It's hard to keep track and know that they are in OR mode.